test/test_autograd.py

```diff
@@ -1502,6 +1502,66 @@ def func(x):
         gradcheck(func, [x])
         gradgradcheck(func, [x])
 
+    def test_complex_views(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        x.real.sum().backward()
+        self.assertEqual(x.grad, torch.ones_like(x))
+
+        y = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        y.imag.sum().backward()
+        self.assertEqual(y.grad, -1j * torch.ones_like(y))
+
+        def func(z):
+            z_ = torch.view_as_complex(z)
+            z_select = torch.select(z_, z_.dim() - 1, 0)
+            z_select_real = torch.view_as_real(z_select)
+            return z_select_real.sum()
+
+        z = torch.randn(10, 2, 2, dtype=torch.double, requires_grad=True)
+        gradcheck(func, [z])
+
+        z1 = z.clone().detach().requires_grad_(True)
+        torch.select(z1, z1.dim() - 2, 0).sum().backward()
+
+        func(z).backward()
+        self.assertEqual(z.grad, z1.grad)
+
+    def test_view_func_for_complex_views(self):
```

Comment:
This might be clearer and more reliable if you separate out the test to test each case one after the other.
Also I think you want to test the following to actually catch errors:

Something like the following (that uses chunk that is currently broken and does not mark the result as views):
```python
import torch

a = torch.rand(10, dtype=torch.double, requires_grad=True)

def fn(a):
    b = a.clone()
    c1 = b.chunk(4)[0]
    c2 = b.chunk(4)[0]

    c1.mul_(2)
    c2.mul_(3)

    return b

torch.autograd.gradcheck(fn, a)
```

Reply:
so you mean replace `chunk` with `view_as_complex` and `view_as_real` and add a similar test?

Reply:
Yes change the way c1 and c2 are created to use the view function you want to test.
This should fail before this PR and work with it.